tests/hash_table_test.c

```diff
@@ -0,0 +1,411 @@
+/*
+* Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/hash_table.h>
+#include <aws/testing/aws_test_harness.h>
+#include <stdio.h>
+#include <sys/time.h>
+
+static const char *test_str_1 = "test 1";
+static const char *test_str_2 = "test 2";
+
+static const char *test_val_str_1 = "value 1";
+static const char *test_val_str_2 = "value 2";
+
+AWS_TEST_CASE(test_hash_table_put_get, test_hash_table_put_get_fn)
+static int test_hash_table_put_get_fn(struct aws_allocator *alloc, void *ctx) {
+    struct aws_common_hash_table hash_table;
+    int err_code = aws_common_hash_table_init(&hash_table, alloc, 10, aws_common_hash_string,
+        aws_common_string_eq, NULL);
+    struct aws_common_hash_element *pElem;
+    int was_created;
+
+    ASSERT_INT_EQUALS(AWS_ERROR_SUCCESS, err_code,
```

Comment:
Once you fix the raise error issue these would all be ASSERT_SUCCESS AND ASSERT_ERROR